packages/reakit/src/Composite/CompositeState.ts

```diff
@@ -579,6 +579,17 @@ function useAction<T extends (...args: any[]) => any>(fn: T) {
   return React.useCallback(fn, []);
 }
 
+function useIsMountedRef() {
+  const isMountedRef = React.useRef(true);
+  React.useLayoutEffect(
+    () => () => {
+      isMountedRef.current = false;
```

Comment:
I think I may be misunderstanding something but the conventions of the ref seem backwards to me.

We set isMounted = true, and then when react renders to the DOM for the first time (mound) we set it to false.

It would be great if you could add some comments to explain, unless what I'm missing/misunderstanding is painfully obvious to anyone who hasn't just woken up on a Monday morning!

Reply:
It's setting isMounted to `false` in the return function of the effect callback. Maybe `isUnmountedRef` would make more sense?

Reply:
Added an explicit `return` statement, so hopefully it's more clear:
```js
function useIsUnmountedRef() {
  const isUnmountedRef = React.useRef(false);
  useIsomorphicEffect(() => {
    return () => {
      isUnmountedRef.current = true;
    };
  }, []);
  return isUnmountedRef;
}
```

Reply:
That is exactly what I was missing, makes sense!